--- setools-3.3.8/python/setools/Makefile.am~	2013-01-16 17:36:22.000000000 +0100
+++ setools-3.3.8/python/setools/Makefile.am	2016-05-05 11:07:24.558491996 +0200
@@ -25,7 +25,7 @@
 	LIBS="$(QPOL_LIB_FLAG) $(APOL_LIB_FLAG)" INCLUDES="$(QPOL_CFLAGS) $(APOL_CFLAGS)" $(PYTHON) setup.py build
 
 install-exec-hook:
-	$(PYTHON) setup.py install `test -n "$(DESTDIR)" && echo --root $(DESTDIR)`
+	$(PYTHON) setup.py install --prefix=$(prefix) `test -n "$(DESTDIR)" && echo --root $(DESTDIR)`
 
 uninstall-hook: 
 	$(PYTHON) setup.py uninstall `test -n "$(DESTDIR)" && echo --root $(DESTDIR)`
