--- setools-3.3.8/configure.ac.orig	2015-10-26 17:06:19.871423815 +0100
+++ setools-3.3.8/configure.ac	2015-10-26 17:08:34.501753575 +0100
@@ -412,13 +412,13 @@
 sepol_devel_incdir="${sepol_devel}/include"
 selinux_devel_incdir="${selinux_devel}/include"
 dnl if /lib64 exists then use that directory, otherwise revert to just /lib
-for dir in lib64 lib ; do
+for dir in libx32 lib64 lib ; do
   sepol_devel_libdir="${sepol_devel}/${dir}"
   if test -f ${sepol_devel_libdir}/libsepol.so ; then
     break
   fi
 done
-for dir in lib64 lib ; do
+for dir in libx32 lib64 lib ; do
   selinux_devel_libdir="${selinux_devel}/${dir}"
   if test -f ${selinux_devel_libdir}/libselinux.so ; then
     break
