--- setools-3.3.8/libseaudit/src/bool_message.c.orig	2013-01-16 17:36:23.000000000 +0100
+++ setools-3.3.8/libseaudit/src/bool_message.c	2014-03-05 10:48:42.193733896 +0100
@@ -101,7 +101,7 @@
 		return NULL;
 	}
 	if ((misc_string = bool_message_to_misc_string(boolm)) == NULL ||
-	    apol_str_appendf(&s, &len, misc_string) < 0 || apol_str_append(&s, &len, close_brace) < 0) {
+	    apol_str_appendf(&s, &len, "%s%s", misc_string, close_brace) < 0) {
 		free(misc_string);
 		return NULL;
 	}
@@ -128,7 +128,7 @@
 		return NULL;
 	}
 	if ((misc_string = bool_message_to_misc_string(boolm)) == NULL ||
-	    apol_str_appendf(&s, &len, misc_string) < 0 || apol_str_appendf(&s, &len, "%s%s<br>", s, close_brace) < 0) {
+	    apol_str_appendf(&s, &len, "%s%s%s<br>", misc_string, s, close_brace) < 0) {
 		free(misc_string);
 		return NULL;
 	}
--- setools-3.3.8/libseaudit/src/filter.c.orig	2013-01-16 17:36:23.000000000 +0100
+++ setools-3.3.8/libseaudit/src/filter.c	2014-03-05 10:49:54.833730849 +0100
@@ -1108,8 +1108,7 @@
 	if ((file = fopen(filename, "w")) == NULL) {
 		return -1;
 	}
-	fprintf(file, XML_VER);
-	fprintf(file, "<view xmlns=\"http://oss.tresys.com/projects/setools/seaudit-%s/\">\n", FILTER_FILE_FORMAT_VERSION);
+	fprintf(file, "%s<view xmlns=\"http://oss.tresys.com/projects/setools/seaudit-%s/\">\n", XML_VER, FILTER_FILE_FORMAT_VERSION);
 	filter_append_to_file(filter, file, 1);
 	fprintf(file, "</view>\n");
 	fclose(file);
--- setools-3.3.8/libseaudit/src/model.c.orig	2013-01-16 17:36:23.000000000 +0100
+++ setools-3.3.8/libseaudit/src/model.c	2014-03-05 10:50:24.457062938 +0100
@@ -514,8 +514,8 @@
 	if ((file = fopen(filename, "w")) == NULL) {
 		return -1;
 	}
-	fprintf(file, XML_VER);
-	fprintf(file, "<view xmlns=\"http://oss.tresys.com/projects/setools/seaudit-%s/\" name=\"%s\" match=\"%s\" show=\"%s\">\n",
+	fprintf(file, "%s<view xmlns=\"http://oss.tresys.com/projects/setools/seaudit-%s/\" name=\"%s\" match=\"%s\" show=\"%s\">\n",
+		XML_VER,
 		FILTER_FILE_FORMAT_VERSION, model->name,
 		model->match == SEAUDIT_FILTER_MATCH_ALL ? "all" : "any",
 		model->visible == SEAUDIT_FILTER_VISIBLE_SHOW ? "true" : "false");
--- setools-3.3.8/sediff/progress.c.orig	2013-01-16 17:36:24.000000000 +0100
+++ setools-3.3.8/sediff/progress.c	2014-03-05 10:57:45.037044450 +0100
@@ -115,10 +115,10 @@
 	}
 	g_mutex_unlock(progress->mutex);
 	if (progress->done < 0) {
-		toplevel_ERR(progress->top, GTK_LABEL(progress->label2)->label);
+		toplevel_ERR(progress->top, "%s", GTK_LABEL(progress->label2)->label);
 		return progress->done;
 	} else if (progress->done > 1) {
-		toplevel_WARN(progress->top, GTK_LABEL(progress->label2)->label);
+		toplevel_WARN(progress->top, "%s", GTK_LABEL(progress->label2)->label);
 		return progress->done - 1;
 	} else {
 		progress->done = 0;
--- setools-3.3.8/sediff/toplevel.c.orig	2013-01-16 17:36:24.000000000 +0100
+++ setools-3.3.8/sediff/toplevel.c	2014-03-05 11:04:39.027027078 +0100
@@ -453,7 +453,7 @@
 		ERR(NULL, "%s", strerror(errno));
 		return;
 	}
-	dialog = gtk_message_dialog_new(top->w, GTK_DIALOG_DESTROY_WITH_PARENT, msg_type, GTK_BUTTONS_CLOSE, msg);
+	dialog = gtk_message_dialog_new(top->w, GTK_DIALOG_DESTROY_WITH_PARENT, msg_type, GTK_BUTTONS_CLOSE, "%s", msg);
 	free(msg);
 	gtk_dialog_run(GTK_DIALOG(dialog));
 	gtk_widget_destroy(dialog);
--- setools-3.3.8/sediff/utilgui.c.orig	2013-01-16 17:36:24.000000000 +0100
+++ setools-3.3.8/sediff/utilgui.c	2014-03-05 11:06:56.727021298 +0100
@@ -31,7 +31,7 @@
 void util_message(GtkWindow * parent, GtkMessageType msg_type, const char *msg)
 {
 	GtkWidget *dialog;
-	dialog = gtk_message_dialog_new(parent, GTK_DIALOG_DESTROY_WITH_PARENT, msg_type, GTK_BUTTONS_CLOSE, msg);
+	dialog = gtk_message_dialog_new(parent, GTK_DIALOG_DESTROY_WITH_PARENT, msg_type, GTK_BUTTONS_CLOSE, "%s", msg);
 	gtk_dialog_run(GTK_DIALOG(dialog));
 	gtk_widget_destroy(dialog);
 }
--- setools-3.3.8/seaudit/progress.c.orig	2013-01-16 17:36:24.000000000 +0100
+++ setools-3.3.8/seaudit/progress.c	2014-03-05 11:23:30.663646251 +0100
@@ -114,10 +114,10 @@
 	}
 	g_mutex_unlock(progress->mutex);
 	if (progress->done < 0) {
-		toplevel_ERR(progress->top, GTK_LABEL(progress->label2)->label);
+		toplevel_ERR(progress->top, "%s", GTK_LABEL(progress->label2)->label);
 		return progress->done;
 	} else if (progress->done > 1) {
-		toplevel_WARN(progress->top, GTK_LABEL(progress->label2)->label);
+		toplevel_WARN(progress->top, "%s", GTK_LABEL(progress->label2)->label);
 		return progress->done - 1;
 	} else {
 		progress->done = 0;
--- setools-3.3.8/seaudit/toplevel.c.orig	2013-01-16 17:36:24.000000000 +0100
+++ setools-3.3.8/seaudit/toplevel.c	2014-03-05 11:23:58.290311759 +0100
@@ -902,7 +902,7 @@
 		ERR(NULL, "%s", strerror(errno));
 		return;
 	}
-	dialog = gtk_message_dialog_new(top->w, GTK_DIALOG_DESTROY_WITH_PARENT, msg_type, GTK_BUTTONS_CLOSE, msg);
+	dialog = gtk_message_dialog_new(top->w, GTK_DIALOG_DESTROY_WITH_PARENT, msg_type, GTK_BUTTONS_CLOSE, "%s", msg);
 	free(msg);
 	gtk_dialog_run(GTK_DIALOG(dialog));
 	gtk_widget_destroy(dialog);
--- setools-3.3.8/seaudit/utilgui.c.orig	2013-01-16 17:36:24.000000000 +0100
+++ setools-3.3.8/seaudit/utilgui.c	2014-03-05 11:24:33.873643600 +0100
@@ -30,7 +30,7 @@
 void util_message(GtkWindow * parent, GtkMessageType msg_type, const char *msg)
 {
 	GtkWidget *dialog;
-	dialog = gtk_message_dialog_new(parent, GTK_DIALOG_DESTROY_WITH_PARENT, msg_type, GTK_BUTTONS_CLOSE, msg);
+	dialog = gtk_message_dialog_new(parent, GTK_DIALOG_DESTROY_WITH_PARENT, msg_type, GTK_BUTTONS_CLOSE, "%s", msg);
 	gtk_dialog_run(GTK_DIALOG(dialog));
 	gtk_widget_destroy(dialog);
 }
