diff -ur setools.org/libapol/Makefile setools.new/libapol/Makefile
--- setools.org/libapol/Makefile	2006-01-23 16:58:42.000000000 +0100
+++ setools.new/libapol/Makefile	2006-04-17 00:31:08.188294750 +0200
@@ -21,7 +21,7 @@
 CFLAGS  += -DLIBAPOL_POLICY_INSTALL_DIR='"$(POLICY_INSTALL_DIR)"'
 CFLAGS  += -DLIBAPOL_SELINUX_DIR='"$(SELINUX_DIR)"'
 CFLAGS	+= -DLIBAPOL_DEFAULT_POLICY='"$(POLICY_SRC_FILE)"'
-CFLAGS  += -I.
+CFLAGS  += -I. -fPIC
 
 LDFLAGS += -lfl
 
diff -ur setools.org/libseaudit/Makefile setools.new/libseaudit/Makefile
--- setools.org/libseaudit/Makefile	2006-01-23 16:58:42.000000000 +0100
+++ setools.new/libseaudit/Makefile	2006-04-17 00:32:42.678200000 +0200
@@ -4,6 +4,8 @@
 SEAUDIT_LIBS   = libseaudit.a ../libapol/libapol.a $(LIBS)
 CFLAGS		+= -D_GNU_SOURCE
 CFLAGS          += -DLIBSEAUDIT_VERSION_STRING='"$(shell cat VERSION)"'
+CFLAGS		+= -fPIC
+LDFLAGS	+= -lxml2
 LIBXML_FLAGS   = `pkg-config --cflags libxml-2.0`
 
 TARGET = libseaudit.so
diff -ur setools.org/libsefs/fsdata.c setools.new/libsefs/fsdata.c
--- setools.org/libsefs/fsdata.c	2006-01-23 16:58:42.000000000 +0100
+++ setools.new/libsefs/fsdata.c	2006-04-17 00:30:06.880463250 +0200
@@ -10,7 +10,7 @@
 
 #include "fsdata.h"
 /* sqlite db stuff */
-#include "sqlite/sqlite3.h"
+#include <sqlite3.h>
 
 /* SE Linux includes*/
 #include <selinux/selinux.h>
diff -ur setools.org/libsefs/Makefile setools.new/libsefs/Makefile
--- setools.org/libsefs/Makefile	2006-01-23 16:58:42.000000000 +0100
+++ setools.new/libsefs/Makefile	2006-04-17 00:29:01.912403000 +0200
@@ -25,10 +25,10 @@
 	$(MAKE) -C sqlite all
 
-$(LIBSO): sqlite $(LIB-OBJ)  
-	$(CC) $(LDFLAGS) -shared -o $(LIBSO) $(LIB-OBJ) $(SQL-OBJ) -Wl,-soname,$(SONAME) 
+$(LIBSO): $(LIB-OBJ)  
+	$(CC) $(LDFLAGS) -shared -o $(LIBSO) $(LIB-OBJ) -Wl,-soname,$(SONAME) -lsqlite3
 
-libsefs.a: sqlite $(LIB-OBJ) 
-	 ar cr $@ $(LIB-OBJ) $(SQL-OBJ)
+libsefs.a: $(LIB-OBJ) 
+	 ar cr $@ $(LIB-OBJ)
 
 install-libsefs-shared: $(LIBSO) install-includes
 	install -m 755 $(LIBSO) $(SHARED_LIB_INSTALL_DIR)
diff -ur setools.org/Makefile setools.new/Makefile
--- setools.org/Makefile	2006-01-23 16:58:42.000000000 +0100
+++ setools.new/Makefile	2006-04-17 00:24:58.009160000 +0200
@@ -89,9 +89,9 @@
 endif
 
 ifeq ($(DEBUG), 0)
-CFLAGS			+= -Wall -O2 -fPIC $(CC_DEFINES)
+CFLAGS			+= $(OPT) -Wall -fPIC $(CC_DEFINES)
 else
-CFLAGS			+= -Wall -g -fPIC $(CC_DEFINES) -DDEBUG
+CFLAGS			+= $(OPT) -Wall -fPIC $(CC_DEFINES) -DDEBUG
 #CFLAGS			= -Wall -ansi -pedantic -g $(CC_DEFINES)
 endif
 
diff -ur setools.org/secmds/Makefile setools.new/secmds/Makefile
--- setools.org/secmds/Makefile	2006-01-20 23:01:04.000000000 +0100
+++ setools.new/secmds/Makefile	2006-04-17 00:52:47.641505500 +0200
@@ -47,7 +47,7 @@
 	$(CC) -c indexcon.c $(CFLAGS) $(INCLUDE)
 
 indexcon: indexcon.o $(LIBSEFS) $(LIBAPOL)
-	$(CC) -o $@ indexcon.o $(LDFLAGS) $(STATICSEFS) $(LIBSELINUXSO) $(STATICAPOL) $(LIBS)
+	$(CC) -o $@ indexcon.o $(LDFLAGS) $(STATICSEFS) $(LIBSELINUXSO) $(STATICAPOL) $(LIBS) -lsqlite3
 
 searchcon.o: searchcon.c
 	$(CC) -c searchcon.c $(CFLAGS) $(INCLUDE)
