--- setools-3.3.8/configure.ac.orig	2013-01-16 17:36:24.000000000 +0100
+++ setools-3.3.8/configure.ac	2014-03-04 15:44:31.839948212 +0100
@@ -368,9 +368,9 @@
 fi
 
 if test "x${enable_debug}" = xyes; then
-   TCL_AUTOPATH+=' $(top_builddir)/libqpol/swig/tcl $(top_builddir)/libapol/swig/tcl $(top_builddir)/libsefs/swig/tcl $(top_builddir)/libpolsearch/swig/tcl $(top_builddir)/apol'
+   TCL_AUTOPATH="$TCL_AUTOPATH"' $(top_builddir)/libqpol/swig/tcl $(top_builddir)/libapol/swig/tcl $(top_builddir)/libsefs/swig/tcl $(top_builddir)/libpolsearch/swig/tcl $(top_builddir)/apol'
 else
-   TCL_AUTOPATH+=' $(libdir)/setools'
+   TCL_AUTOPATH="$TCL_AUTOPATH"' $(libdir)/setools'
 fi
 AC_SUBST(TCL_AUTOPATH)
 
@@ -688,8 +688,8 @@
 AC_SUBST(SQLITE3_CFLAGS)
 AC_SUBST(SQLITE3_LIBS)
 
-SEAUDIT_LIB_FLAG+=" ${XML_LIBS}"
-SEFS_LIB_FLAG+=" ${SQLITE3_LIBS}"
+SEAUDIT_LIB_FLAG="$SEAUDIT_LIB_FLAG ${XML_LIBS}"
+SEFS_LIB_FLAG="$SEFS_LIB_FLAG ${SQLITE3_LIBS}"
 
 gtk_version_2_8=1
 if test "x${build_gui}" = xyes; then
